### PR TITLE
Droth 496 link attribute information import tool

### PR DESCRIPTION
**Pull request by marquesfm**

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala [hunk 1]
```diff
@@ -389,23 +390,24 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
...
-  }
+//  private def updateExistingLinkPropertyRow(table: String, column: String, linkId: Long, username: Option[String], existingValue: Int, value: Int) = {
+//    if (existingValue != value) {
+//      sqlu"""update #$table
+//               set #$column = $value,
+//                   modified_date = current_timestamp,
+//                   modified_by = $username
+//               where link_id = $linkId""".execute
+//    }
+//  }
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala]: Remove this comment code from here.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala [hunk 2]
```diff
@@ -389,23 +390,24 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
...
 
   protected def setLinkProperty(table: String, column: String, value: Int, linkId: Long, username: Option[String],
                                 optionalVVHValue: Option[Int] = None, latestModifiedAt: Option[String],
                                 latestModifiedBy: Option[String]) = {
-    val optionalExistingValue: Option[Int] = sql"""select #$column from #$table where link_id = $linkId""".as[Int].firstOption
+    //val optionalExistingValue: Option[Int] = sql"""select #$column from #$table where link_id = $linkId""".as[Int].firstOption
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala]: Remove this comment code from here

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala [hunk 3]
```diff
@@ -420,10 +422,11 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
...
                                  latestModifiedBy: Option[String]) = {
     if (latestModifiedAt.isEmpty) {
-      sqlu"""insert into #$table (id, link_id, #$column, modified_by)
-                 select primary_key_seq.nextval, $linkId, $value, $username
-                 from dual
-                 where not exists (select * from #$table where link_id = $linkId)""".execute
+//      sqlu"""insert into #$table (id, link_id, #$column, modified_by)
+//                 select primary_key_seq.nextval, $linkId, $value, $username
+//                 from dual
+//                 where not exists (select * from #$table where link_id = $linkId)""".execute
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala]: Remove this comment code from here.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala [hunk 4]
```diff
@@ -478,24 +481,24 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
...
+//    sql"""select f.link_id, f.functional_class, f.modified_date, f.modified_by
+//            from functional_class f
+//            join #$idTableName i on i.id = f.link_id""".as[(Long, Int, DateTime, String)].list
+//  }
+//
+//  private def fetchLinkTypes(idTableName: String): Seq[(Long, Int, DateTime, String)] = {
+//    sql"""select l.link_id, l.link_type, l.modified_date, l.modified_by
+//            from link_type l
+//            join #$idTableName i on i.id = l.link_id""".as[(Long, Int, DateTime, String)].list
+//  }
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RoadLinkService.scala]: Removed comment code. It's not used anywhere else.

File: digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/ImportDataApi.scala [hunk 5]
```diff
@@ -0,0 +1,63 @@
+package fi.liikennevirasto.digiroad2.dataimport
+
+import fi.liikennevirasto.digiroad2.Digiroad2Context.{Digiroad2ServerOriginatedResponseHeader, userProvider}
+import fi.liikennevirasto.digiroad2.authentication.RequestHeaderAuthentication
+import fi.liikennevirasto.digiroad2.dataimport.DataCsvImporter.RoadLinkCsvImporter.{ExcludedLink, ImportResult}
```
Comment (marquesfm) [inline digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/ImportDataApi.scala]: ExcludedLink import are not usaged, verify if it is necessary

File: digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/ImportDataApi.scala [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import org.scalatra.servlet.FileUploadSupport
+import org.scalatra.json.JacksonJsonSupport
+
+class ImportDataApi extends ScalatraServlet with FileUploadSupport with JacksonJsonSupport with RequestHeaderAuthentication {
+
+  protected implicit val jsonFormats: Formats = DefaultFormats
+  private val CSV_LOG_PATH = "/tmp/csv_data_import_logs/"
+  private val  ROAD_LINK_LOG = "road link import"
+  private val csvImporter = new RoadLinkCsvImporter {
+   // override val roadLinkService: RoadLinkService = Digiroad2Context.roadLinkService
```
Comment (marquesfm) [inline digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/ImportDataApi.scala]: commented code, need to be removed

File: digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporterSpec.scala [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    }
+  }
+}
+
+class CsvDataImporterSpec extends AuthenticatedApiSpec with BeforeAndAfter {
+  val MunicipalityKauniainen = 235
+  val testUserProvider = new OracleUserProvider
+  val roadLinkCsvImporter = importerWithNullService()
+
+  private def importerWithService(testVVHClient: VVHClient) : RoadLinkCsvImporter = {
```
Comment (marquesfm) [inline digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporterSpec.scala]: method is never used, please removed

**Comment by joanacoelho** [comment]
ok

File: digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporter.scala [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+  private val textFieldMappings = Map(
+    "Tien nimi (suomi)" -> "ROADNAME_FI",
+    "Tien nimi (ruotsi)" -> "ROADNAME_SE",
+    "Tien nimi (saame)" -> "ROADNAME_SM"
+  )
+
+  private val mandatoryFields = "Linkin ID"
+
+
```
Comment (oskarmalmstrom) [inline digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporter.scala]: unnecessary line break

**Comment by marquesfm** [comment]
ok for me

**Comment by halied** [comment]
Ok

**Comment by marquesfm** [comment]
OK

**Pull request merged**